app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -524,6 +522,33 @@ def traces_helpers(search:)
         ::Katello::HostTracer.helpers_for(traces)
       end
 
+      def probably_rhel?
+        # Get the os name from sub-man facts rather than operatingsystem. This is
+        # less likely to have been changed by the user.
+        os_name = facts['distribution::name']
+        # if this fact isn't there, we can ignore it because the host is not "managed"
+        os_name.present? && os_name.start_with?('Red Hat Enterprise Linux')
+      end
+
+      def rhel_eos_schedule_index
+        return nil unless probably_rhel?
+        major = operatingsystem&.major
+        arch_name = architecture&.name
+        return "RHEL#{major}" unless major == "7"
```

Comment:
I'm not 100% sure if `architecture` triggers a foreign key load, but I'd swap the order just in case.
```suggestion
        return "RHEL#{major}" unless major == "7"
        arch_name = architecture&.name
```